frontend/dockerfile/parser/split_command.go

```diff
@@ -55,32 +57,44 @@ func extractBuilderFlags(line string) (string, []string, error) {
 				continue
 			}
 
-			// Only keep going if the next word starts with --
-			if ch != '-' || pos+1 == len(line) || rune(line[pos+1]) != '-' {
+			// Only keep going if the next word starts with -- or if we're parsing a flag argument
+			if !parsingFlagArg && (ch != '-' || pos+1 == len(line) || rune(line[pos+1]) != '-') {
```

Comment:
Is `run(line[pos+1])` (specifically the `+1` bit) valid in the possible presence of multibyte characters in the line? Something built around `strings.HasPrefix("--", line[pos])` might be safer? OTOH I'm not convinced the existing code is safe in that way -- I guess it assumes all commands and flag names are 7 bit ASCII and hence safe to operate over as bytes on a UTF-8 stream, a `text/scanner.Scanner` for the outer loop might be safer (but a much bigger change).

Reply:
I'm not modifying the existing behavior, this is not meant to be a refactor PR.

Reply:
Wow, `parseWords` and `extractBuilderFlags` are _so_ close. Except the former handles the unicode case correctly...